### PR TITLE
RecyclerView and Adapter

### PR DESCRIPTION
**Pull request by nttung291**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -29,4 +29,6 @@ dependencies {
     implementation 'com.android.support:appcompat-v7:26.1.0'
     implementation 'com.android.support:design:26.1.0'
     implementation 'com.android.support:recyclerview-v7:26.1.0'
+    compile 'com.android.support:cardview-v7:26.1.0'
```
Comment (DoanVanToan) [inline app/build.gradle]: e nên sử dụng implementation

File: app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+public class MovieAdapter extends BaseRecyclerViewAdapter<MovieAdapter.ViewHolder> {
+    private ArrayList<Movie> mMovies = new ArrayList<>();
+    private LayoutInflater layoutInflater;
+
+    public MovieAdapter(@NonNull Context context) {
+        super(context);
+    }
+
+    public void replaceData(@NonNull List<Movie> movies) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java]: check null movies

File: app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.framgia.moviedb.R;
+import com.framgia.moviedb.model.Movie;
+import com.framgia.moviedb.screen.BaseRecyclerViewAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MovieAdapter extends BaseRecyclerViewAdapter<MovieAdapter.ViewHolder> {
+    private ArrayList<Movie> mMovies = new ArrayList<>();
+    private LayoutInflater layoutInflater;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java]: check convention

File: app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        holder.bindData(mMovies.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mMovies != null ? mMovies.size() : 0;
+    }
+
+    class ViewHolder extends RecyclerView.ViewHolder {
+        private ImageView ivPosterFilm;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/adapter/MovieAdapter.java]: check convention

File: app/src/main/java/com/framgia/moviedb/model/Movie.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    };
+    private int mId;
+    private String mTitle;
+    private String mOverview;
+    private double mVoteAverage;
+    private String mPosterPath;
+    private String mBackdropPath;
+    private String mReleaseDate;
+
+    public Movie(int id, String title, String overview, double voteAverage, String posterPath,
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/moviedb/model/Movie.java]: delete if unused

Comment (nttung291) [reply app/src/main/java/com/framgia/moviedb/model/Movie.java]: Get Data em co dung a

**Pull request merged**